### PR TITLE
Cars details

### PR DESCRIPTION
**Pull request by zakiyousufi**
### In this PR I have created the cars details page

### CODE REVIEW

**Review by Shisui6 — CHANGES_REQUESTED** [review]
Hi @zakiyousufi, 

# Requires Changes 🍀
Nice work!👏👏
There are just a few issues you need to work on🪲

## Highlights
- Details page created ✅

## Required Changes ♻️

- Please link the cars page and the details page using the book icon on the cars page
- Kindly add your details component to the react-router in main.jsx
- Kindly update your package.json file to include react icons
- Kindly update your details page to always fetch the correct data from the API. Please also include a condition for when the requested car does not exist in the database

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

You can also consider the following:

N/A

Cheers and Happy coding!🍻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

File: src/main.jsx [hunk 1]
```diff
@@ -38,6 +39,13 @@ const router = createBrowserRouter([
         element:
   <RequireAuth loginPath="/login">
     <Cars />
+  </RequireAuth>,
+      },
+      {
+        path: 'car',
```
Comment (Shisui6) [inline src/main.jsx]: Kindly update the path here to include the id of the car
```suggestion
        path: 'cars/:id',
```

File: src/components/Cars/Cars.jsx [hunk 2]
```diff
@@ -80,7 +80,7 @@ const Cars = () => {
                   </p>
                   <div className="mt-4 flex justify-center gap-4 text-gray-400">
                     <Link
-                      to="/"
+                      to="/car"
```
Comment (Shisui6) [inline src/components/Cars/Cars.jsx]: Kindly update this path to include the id of the car in the URL params
```suggestion
                      to={`/cars/${car.id}`}
```

File: src/components/Details/carsDetails.jsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
+import React, { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { useAuthHeader } from 'react-auth-kit';
+import { useParams, useDispatch } from 'react-router-dom';
```
Comment (Shisui6) [inline src/components/Details/carsDetails.jsx]: `useDispatch` comes from react-redux not react-router-dom. Please move the import of useDispatch to come from the correct library

File: src/components/Details/carsDetails.jsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+import Loader from '../Loader/Loader';
+import { fetchCars } from '../../redux/cars/cars';
+
+const DetailsCarScreen = () => {
+  const [loading, setLoading] = useState(true);
+  const dispatch = useDispatch();
+  const authHeader = useAuthHeader();
+  const { id } = useParams();
+  const carDetails = useSelector((state) => state.cars);
+  const cars = carDetails;
```
Comment (Shisui6) [inline src/components/Details/carsDetails.jsx]: Kindly use object destructuring to get the cars value from the carDetails object
```suggestion
  const { cars } = carDetails;
```

File: src/components/Details/carsDetails.jsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  const carss = cars?.find((c) => c.id === parseInt(id, 10));
```
Comment (Shisui6) [inline src/components/Details/carsDetails.jsx:24]: [Optional] I'm not sure why you put a `?` next to the cars object. Isn't that ruby syntax?

File: src/components/Details/carsDetails.jsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    );
+  }
+
+  return (
+    <>
+      {carss && (
+        <div className="flex flex-col items-center md:justify-start justify-center w-full md:flex-row grow h-full lg:pt-20 lg:pb-10">
+          <div className="grow  md:w-5/6 flex items-center justify-center md:px-10 rounded-full aspect-square">
+            <img
+              src={carss.image?.url}
```
Comment (Shisui6) [inline src/components/Details/carsDetails.jsx]: The image is stored in `carss.image` not in `carss.image.url`. Please update this line

File: src/components/Details/carsDetails.jsx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+          There is no car in Garage
+        </p>
+
+      </>
+    );
+  }
+
+  return (
+    <>
+      {carss && (
```
Comment (Shisui6) [inline src/components/Details/carsDetails.jsx:51]: Kindly update the design of the details page to match the behance design better🙏

Your design:
image


Behance design:
image

A few things to note:
- Please include more details for the car. The car has a type, model, year, and price_per_day
- Kindly increase the right padding of the details. Its a bit too close to the edge
- Kindly raise the car details to be higher up on the page. Not on the same line with the image
- Kindly increase the spacing between `other details` and `discover more models`
- Kindly add the reserve button on the bottom right of the page. Its okay if it doesn't work for now, at least let it be there
- Kindly add the back button on the bottom left of the page. it was there before but you removed it

**Review by Shisui6 — CHANGES_REQUESTED** [review]
Hi @zakiyousufi, 

# Requires Changes 🍀
You are doing a great job. The details page is looking great👍
There are just a few more issues to look into🪲

## Highlights
- Details page implemented ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

You can also consider the following:

N/A

Cheers and Happy coding!🍻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

File: src/components/Details/carsDetails.jsx (hunk 5 above)
Comment (zakiyousufi) [reply src/components/Details/carsDetails.jsx:24]: @Shisui6 Yh it's ruby syntax and actully we use it on react as well to ensure that the value is exist on API as we use it on ruby 
car? ===> if (car exist ) then find the id

File: src/components/Details/carsDetails.jsx (hunk 6 above)
Comment (zakiyousufi) [reply src/components/Details/carsDetails.jsx]: Ok i'll do it

**Review by zakiyousufi — COMMENTED** [review]
Hello @Shisui6  Thanks for your kind review

**Review by Shisui6 — APPROVED** [review]
Hi @zakiyousufi, 

# Approved 🪄
Your project is complete! There is nothing else to say other than... it's time to merge it 💎
Congratulations! 🎉 

## Highlights
- Everything works fine ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will really improve your application._

You can also consider the following:

N/A

Cheers and Happy coding!🍻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Review by Sonickmumba — APPROVED** [review]
Hi @zakiyousufi , 

Well done. Your project is complete!  There is nothing else to say other than... it's time to merge it: 
Congratulations! 🎉 

## Highlights 💪 
- No linter errors
- project requirements are met
- project works fine
- good PR description
- correct flow is used

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Sonickmumba  in your question so I can receive the notification.

------

**Pull request merged**